presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java

```diff
@@ -318,10 +318,8 @@ else if (leftNDV <= matchingRightNDV) {
             return UNKNOWN_STATS;
         }
 
-        // account for remaining clauses
-        for (int i = 0; i < remainingClauses.size(); ++i) {
-            result = result.mapOutputRowCount(rowCount -> min(leftStats.getOutputRowCount(), rowCount / UNKNOWN_FILTER_COEFFICIENT));
-        }
+        // limit the number of complement rows (to left row count) and account for remaining clauses
+        result = result.mapOutputRowCount(rowCount -> min(leftStats.getOutputRowCount(), rowCount / Math.pow(UNKNOWN_FILTER_COEFFICIENT, numberOfRemainingClauses)));
```

Comment:
This also affects equijoins with 3+ criteria.  Did you notice any effect on those kinds of queries?

Reply:
@rschlussel2  You mean that for 3 criteria the computed `result` will be different?

Reply:
yes

Reply:
> This also affects equijoins with 3+ criteria. Did you notice any effect on those kinds of queries?

The code treats equicriteria the same as it was before this change. I just unified how non-equi criteria are treated.